src/extension.ts

```diff
@@ -92,14 +88,11 @@ function setLanguageClient(languageClient: vslc.LanguageClient = null) {
     g_languageClient = languageClient
 }
 
-const g_onDidChangeConfig = new vscode.EventEmitter<settings.ISettings>()
+const g_onDidChangeConfig = new vscode.EventEmitter<vscode.ConfigurationChangeEvent>()
 export const onDidChangeConfig = g_onDidChangeConfig.event
-function changeConfig(params: vscode.ConfigurationChangeEvent) {
-    const newSettings = settings.loadSettings()
-    g_onDidChangeConfig.fire(newSettings)
-
-    const need_to_restart_server = g_settings.juliaExePath !== newSettings.juliaExePath ? true : false
-    if (need_to_restart_server) {
+function changeConfig(event: vscode.ConfigurationChangeEvent) {
+    g_onDidChangeConfig.fire(event)
+    if (event.affectsConfiguration('julia.executablePath')) {

```

Comment:
Should this also restart the REPL session?

Reply:
hmm, killing/starting REPL session sounds to me like a big deal, so I guess some users will complain on that, maybe ? Not sure tbh, though.

Reply:
True, I wouldn't want that change to mess up my current session.